[24539] Added documentation for xy charts (#300)
eProsima/Fast-DDS-monitor: docs/rst/user_manual/pro/xy_charts.rst

.. include:: ../../exports/alias.include
.. include:: ../../exports/roles.include

.. _xy_charts:

#################
XY Charts  |Pro|
#################

An *XY Chart* (also called a *phase plot* or *scatter chart*) plots the relationship between two numeric
fields extracted from live DDS topic samples.
Unlike a Topic Chart, whose X axis is always a time axis, an XY Chart has both axes as value axes: the X
axis is driven by one numeric field and the Y axis by another.
New data points appear in real time as samples arrive from the monitored DDS network.

Multiple series can be plotted in the same XY Charts pane. Each series independently sources its X and Y
values, and the two values may come from the same topic or from different topics within the same DDS domain.
Fields that are integers, floats, or doubles can be plotted. Struct and array fields are not plotted
directly but can be expanded to reach their numeric leaf fields.

.. _xy_charts_creating:

Opening an XY Charts
===================

There are several ways to open a new XY Charts pane:

> * Use **Add XY Charts** in the application menu bar.

Reviewer: ```suggestion
* Use **Add Topic Live Chart** in the application menu bar.
```

> * Use the :ref:`right_pane_config` sidebar. When the creation form is open, switch the plot mode to

Reviewer: ```suggestion
* Use the :ref:`right_pane_config` sidebar. When the New Topic Chart form is open, switch the plot mode to
```

  **XY Chart** to switch from a time-based topic chart to an XY scatter chart.

* Click the three-dots button in the header of any existing pane to open the split menu, then choose

>   **Split right** or **Split down** and select **XY Charts** to open a new chart alongside the current

Reviewer: ```suggestion
  **Split right** or **Split down** and select **Topic Chart** to open a new chart alongside the current
```

  pane.

.. _xy_charts_series:

Managing Series
===============

**Adding a series:**

Click **+ Add XY Series** in the **Series** section to expand the series creation form.
For each new series, select:

* An **X Axis Topic** and an **X Field** — the topic and numeric field whose values drive the X axis.

> * A **Y Axis Topic** and a **Y Field** — the topic and numeric field whose values drive the Y axis.

Reviewer: ```suggestion
* An **X Axis Topic** and an **X Field**, the topic and numeric field whose values drive the X axis.
* A **Y Axis Topic** and a **Y Field**, the topic and numeric field whose values drive the Y axis.
```

The same options are available from the **Series** section of the :ref:`right_pane_config` sidebar, where
each series has a row with a color swatch, rename, visibility toggle, and remove button.

**Bulk actions:**

The **Actions** section of the :ref:`right_pane_config` sidebar provides:

* **Show All Series** makes every hidden series visible at once.
* **Hide All Series** hides all series without removing them.
* **Clear Chart** removes all series and resets the chart.

.. _xy_charts_controls:

Chart Header Controls
=====================

The chart header provides the following buttons from left to right:

* |add_series| **Add Series** opens the inline add-series panel in the :ref:`right_pane_config` sidebar.

* |resize| **Reset View** returns both axes to their auto-fit range calculated from the current data.

* |legend| **Toggle Legend** shows or hides the legend listing all active series and their colors.

* |pause| / |play| **Pause / Resume** pauses live data ingestion, freezing the chart at its current state.
  Samples received while paused are discarded. Resuming restarts the live feed from the current time.

> * |info| **Interactive Controls Info** opens a dialog summarising the mouse and keyboard shortcuts

Reviewer: ```suggestion
* |help| **Interactive Controls Info** opens a dialog summarising the mouse and keyboard shortcuts
```

>   available in the chart area.

Reviewer: Remember to change this to reflect the new PanelHelpButton button

* **Apply & Reset Chart** applies any changes to these fields and clears all existing series. This button
  is disabled when no domain is selected.

**Chart Name**

A text field and confirm button to rename the chart title shown in the pane header.

**Display**

* **Show legend** toggles the chart legend on or off.
* **Show lines** connects successive scatter points with a line in addition to the scatter markers.
* **Running** when on, the chart accepts new data in real time. When off, the chart is paused and the
  view stays fixed; resuming restarts the live feed.

**Series**

Lists every series currently added to the chart. Each row shows the series name, a color swatch to change
the marker color, a visibility toggle, and a remove button. An **Add XY Series** button at the bottom
expands the inline series-creation section described below.

**Add XY Series** *(collapsible)*

* **X Axis Topic** — filter and select the topic that provides the X values.
* **X Field** — filter and select the numeric leaf field within that topic.
* **Y Axis Topic** — filter and select the topic that provides the Y values.

> * **Y Field** — filter and select the numeric leaf field within that topic.

Reviewer: ```suggestion
* **X Axis Topic**, filter and select the topic that provides the X values.
* **X Field**, filter and select the numeric leaf field within that topic.
* **Y Axis Topic**, filter and select the topic that provides the Y values.
* **Y Field**, filter and select the numeric leaf field within that topic.
```

> * Click **Add** to create the series. The picker remains open for adding further series.

Reviewer: ```suggestion
* Click **Add XY Series** to create the series. The picker remains open for adding further series.
```
